integration/messaging/test/utils/getFCMToken.js

```diff
@@ -15,15 +15,13 @@
  */
 
 module.exports = async webdriver => {
-  console.log('Getting token....');
+  console.log('Getting token...');
   await webdriver.wait(() => {
     return webdriver.executeScript(() => {
-      return !!window.__test && !!window.__test.token;
+      return !!window.__test;
     });
   });
-
-  console.log('Found token.');
   return webdriver.executeScript(() => {
-    return window.__test.token;
+    return window.__test.triggerGetToken();
```

Comment:
Does this work?

`triggerGetToken()` returns a promise and I'm not sure `executeScript` can pass the browsers promise to node.

Reply:
it works in the sense that the network request is made, and it doesn't fail - which is mostly what I wanted to cover, but agreed that the logic doesn't make much sense... Changing to your suggestion.